File: 2022/tom/day-6/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
)


func main() {
	data := readFile("../test.txt")
	step1 := step1(data)
	fmt.Println(step1)
	step2 := step2(data)
	fmt.Println(step2)
}

func readFile(fileName string) []string {
	data, _ := os.Open(fileName)
	scanner := bufio.NewScanner(data)

	var dataLines []string
	for scanner.Scan() {
		line := scanner.Text()
		dataLines = append(dataLines, line)
	}
	return dataLines
}

func step1(data []string) int {
	r,err := regexp.MatchString(`^.*(.).*\1.*$`,"abcdd")
	fmt.Println(r)
	fmt.Println(err)
	return 0
}

func step2(data []string) int {
	
	return 0
}

File: 2022/tom/day-6/main.js
const { match } = require('assert');
const {readFileSync} = require('fs');

function main() {
  const data = readFile();
  const step1Result = step1(data);
  console.log(step1Result);
  const step2Result = step2(data);
  console.log(step2Result);
}

function readFile() {
  const contents = readFileSync('input.txt', 'utf-8');
  const data = contents.split('\r\n');
  return data;
}

function step1(data)
{
  const r = /^.*(.).*\1.*$/;
  const line = data[0];
  for (i = 0; i < line.length; i++) {
    const sample = line.slice(i, i+4);
    if (!sample.match(r)) {
      return i+4;
    } 
  }
  
  return "";
}

function step2(data)
{
  const r = /^.*(.).*\1.*$/;
  const line = data[0];
  for (i = 0; i < line.length; i++) {
    const sample = line.slice(i, i+14);
    if (!sample.match(r)) {
      return i+14;
    } 
  }
  
  return "";
}

main()